patches/server/0928-More-vanilla-friendly-methods-to-update-trades.patch

```diff
@@ -0,0 +1,74 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sun, 16 Oct 2022 16:12:49 +0200
+Subject: [PATCH] More vanilla friendly methods to update trades
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/npc/Villager.java b/src/main/java/net/minecraft/world/entity/npc/Villager.java
+index 10b45ec24a5a0867106d1694312385ad1e267f43..a12022868ed7c6cb74557bcf1a5028c144035068 100644
+--- a/src/main/java/net/minecraft/world/entity/npc/Villager.java
++++ b/src/main/java/net/minecraft/world/entity/npc/Villager.java
+@@ -919,6 +919,12 @@ public class Villager extends AbstractVillager implements ReputationEventHandler
+ 
+     @Override
+     protected void updateTrades() {
++        // Paper start
++        updateTrades(2);
++    }
++
++    public boolean updateTrades(int amount) {
++        // Paper end
+         VillagerData villagerdata = this.getVillagerData();
+         Int2ObjectMap<VillagerTrades.ItemListing[]> int2objectmap = (Int2ObjectMap) VillagerTrades.TRADES.get(villagerdata.getProfession());
+ 
+@@ -928,9 +934,11 @@ public class Villager extends AbstractVillager implements ReputationEventHandler
+             if (avillagertrades_imerchantrecipeoption != null) {
+                 MerchantOffers merchantrecipelist = this.getOffers();
+ 
+-                this.addOffersFromItemListings(merchantrecipelist, avillagertrades_imerchantrecipeoption, 2);
++                this.addOffersFromItemListings(merchantrecipelist, avillagertrades_imerchantrecipeoption, amount); // Paper
++                return true; // Paper
+             }
+         }
++        return false; // Paper
+     }
+ 
+     public void gossip(ServerLevel world, Villager villager, long time) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftVillager.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftVillager.java
+index f0b910df1ee471b4d72d97c6197ab14f2854976e..187b623b14f177d08793cf90909e516505030b36 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftVillager.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftVillager.java
+@@ -99,6 +99,33 @@ public class CraftVillager extends CraftAbstractVillager implements Villager {
+     }
+ 
+     // Paper start
++    @Override
++    public boolean increaseLevel(int amount) {
++        Preconditions.checkArgument(amount > 0, "Level earned must be positive");
++        int supposedFinalLevel = this.getVillagerLevel() + amount;
++        Preconditions.checkArgument(1 <= supposedFinalLevel && supposedFinalLevel <= 5, "Final level reached after the donation (%d) must be between [1, 5]".formatted(supposedFinalLevel));
```

Comment:
Instead of `1` and `5`, use `VillagerData.MIN_VILLAGER_LEVEL` and `VillagerData.MAX_VILLAGER_LEVEL`